dune/grid/CpGrid.hpp

```diff
@@ -819,6 +820,149 @@ namespace Dune
             }
             return zz/nv;
         }
+
+        const Vector faceCenterEcl(int cell_index, int faceTag) const
+        {
+            // This method is an alternative to the method faceCentroid(...).
+            // The face center is computed as a raw average of cell corners.
+            // For faulted cells this gives different results then average of face nodes
+            // that seems to agree more with eclipse.
+            // This assumes the cell nodes are ordered
+            // 0--1
+            // |  |
+            // 2--3
+            //   4--5
+            //   |  |
+            //   6--7
```

Comment:
@totto82: Please revise the switch case below such that the DUNE Reference Element is used. This allows to exactly extract this kind of information. I have checked the numbers and they are identical.